### ISSUE TITLE
[Bug] Unable to reconnect after a long disconnection

### ISSUE THREAD

**Issue by librehat** [t0bst4r/home-assistant-matter-hub#111]
Just noticed that after I stopped the process by pressing Ctrl+C, then I added it as a systemd service and started in that way (running as the same user).

I noticed that my previously configured bridge is there in the web dashboard, but the "Commissioned" stamp is gone, and my SmartThings app says the device is disconnected. I ran the diagnostics tool from SmartThings but it didn't really help. I had to remove the device from SmartThings, then re-add the bridge via the QR code.

I did notice, however, if I then restart the service via `systemctl restart`, the disconnection didn't happen. Not sure if this was caused because it had been shut down previously for much longer (about 30 minutes), but `systemctl restart` took less than 1 minute

**Comment by t0bst4r** [comment]
No it’s probably caused by some bug related to the storage… That fabrics weren’t persisted somehow.

**Comment by Apollon77** [comment]
or maybe does the systemd looks in a wrong dir? MA<be a cwd needs to be set or such?

**Comment by t0bst4r** [comment]
I have configured the storage service to use:
1. cli parameter `--storage-location <folder>`
2. Environment variable `HAMH=<folder>`
3. User home `~/.home-assistant-matter-hub`

Additionally I have added one fix to the StorageService:
Sometimes JSON files are written with no content. The default Storage Service implementation allows missing files, but does not catch corrupted (empty) files. I catched those and return an empty result:

https://github.com/t0bst4r/home-assistant-matter-hub/blob/main/packages/backend/src/storage/create-storage-service.ts

The app also prints an info-log on boot (one of the first lines).
